src/client/sandbox/event/focus-blur.js

```diff
@@ -181,7 +148,36 @@ export default class FocusBlurSandbox extends SandboxBase {
 
         if (el[type]) {
             // NOTE: To guarantee that all focus/blur events are raised, we need to raise them manually.
-            this._ensureWindowScroll(el, preventScrolling, type, callback);
+            let windowScroll                           = null;
+            let elementNonScrollableParentsScrollState = null;
+
+            if (preventScrolling)
+                windowScroll = styleUtils.getElementScroll(this.window);
+
+            if (browserUtils.isIE)
+                elementNonScrollableParentsScrollState = this._getElementNonScrollableParentsScrollState(el);
+
+            if (this._shouldUseLabelHtmlForElement(el, type)) {
```

Comment:
It is a bad solution too.
You should returns two fields: `el` and flag that equals `it is  label with htmlFor attribute`.
Then we will return is only if it is label with htmlFor attribute and element is null.
So, the current way is simpler.